lib/IRGen/IRGenSIL.cpp

```diff
@@ -8380,9 +8380,17 @@ void IRGenSILFunction::visitClassMethodInst(swift::ClassMethodInst *i) {
   SILDeclRef method = i->getMember().getOverriddenVTableEntry();
   auto methodType = i->getType().castTo<SILFunctionType>();
 
+  AccessLevel methodAccess = method.getDecl()->getEffectiveAccess();
   auto *classDecl = cast<ClassDecl>(method.getDecl()->getDeclContext());
   bool shouldUseDispatchThunk = false;
-  if (IGM.hasResilientMetadata(classDecl, ResilienceExpansion::Maximal)) {
+  // Because typechecking for the debugger has more lax rules, check the access
+  // level of the getter to decide whether to use a dispatch thunk for the
+  // debugger.
+  bool shouldUseDispatchThunkIfInDebugger =
+      !classDecl->getASTContext().LangOpts.DebuggerSupport ||
+      methodAccess == AccessLevel::Public;
+  if (IGM.hasResilientMetadata(classDecl, ResilienceExpansion::Maximal) &&
+      shouldUseDispatchThunkIfInDebugger) {
```

Comment:
Can this be tested by compiling Swift code to IR and passing -debugger-support?

Reply:
I tried that but it's not enough. To look up private classes LLDB implements the `DebuggerClient` interface, without that we can't write a test that can extend a non-public type in a different module, to emulate LLDB's behavior.